index.js

```diff
@@ -31,9 +32,18 @@ const REPORT_INTERVAL = 1000 * 60 // log download progress every minute
  * @param {number} [config.concurrency]
  * @param {number} [config.batchSize]
  */
-export async function startBackup ({ dataURL, s3Region, s3BucketName, s3AccessKeyId, s3SecretAccessKey, s3Endpoint, concurrency, batchSize }) {
+export async function startBackup ({ dataURL, s3Region, s3BucketName, s3AccessKeyId, s3SecretAccessKey, s3Endpoint, concurrency, batchSize, healthcheckPort = 9999 }) {
   const sourceDataFile = dataURL.substring(dataURL.lastIndexOf('/') + 1)
-  const log = debug(`backup:${sourceDataFile}`)
+  const logger = debug(`backup:${sourceDataFile}`)
+  const log = (msg) => {
+    logger(msg)
+    health.updateLastLogged()
+  }
+  const gracePeriod = REPORT_INTERVAL * 2
+  const health = createHealthCheckServer({ sourceDataFile, gracePeriod })
+  health.srv.listen(healthcheckPort, '127.0.0.1', () => {
```

Comment:
is it important to listen on ipv4 vs `'localhost'`?

Reply:
mmm `localhost` on my machine defaults to ipv6 which then causes me issue else where so i'm conditioned to be specific in these things. I know the target machines are not yet configured for ipv6.